### DISCUSSION TITLE
Email templates

### DISCUSSION THREAD

**Discussion by danielboros1990** [nhost/nhost#2829] [General]
Hi guys,

my question is related to email templates. I managed to find the nhost/hasura-auth template. But I have some confusion in case of templates.

1. Is it possible to add more templates apart from the predefined templates?
2. If I can add more templates, is there any api where I can trigger them using nhost functions or an external api?
3. The inbuilt templates have these predefined variables like `link`, `serverUrl`, how is it possible to override these variables, these variables are coming from the nhost/hasura-auth engine, only we can override which are available through the envs?

Thanks for the answers.

**Comment by dbarrosop** [comment]
Hello,
you can refer to our documentation for details on how all of that works:

https://docs.nhost.io/guides/auth/email-templates

Regards

**Reply by danielboros1990** [reply]
I started with this and then I looked through the whole github both for nhost and hasura-auth. The docs is just poorly detailed for the self-hosted version and very though to find solutions for different cases.

**Reply by dbarrosop** [reply]
there is virtually no difference between our cloud and self-hosted when it comes to templates. If you are self-hosting you will need to implement a mechanism to synch your templates with the server where you are running the service but that's purely an infrastructure issue. If you run the help of the Auth service you will see in which path the templates are expected to be located in.

**Reply by danielboros1990** [reply]
@dbarrosop Thanks for you answer, but my 1-2. questions are still unanswered!?

**Reply by dbarrosop** [reply]
1. Not sure what you mean, you can add more locales, you can't add templates for purposes not directly supported by the Auth service.
2. No, refer to my previous comment, you need to implement your own synchronization mechanism if you are self-hosting. Or you can use postmark and then the templates live in their service.

**Reply by neopunisher** [reply]
That link 404s

**Comment by kesavanCgvak** [comment]
Hi @danielboros1990 , I came across your question and I’m looking for the same information. Did you manage to find a solution?
